CMakeLists.txt

```diff
@@ -25,6 +25,49 @@ if(CCACHE_FOUND)
 	set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ccache)
 endif(CCACHE_FOUND)
 
+
+# TA must be built from TA SDK based on GNU Makefiles and not compatible with
+# a CMake build process. Using this macro. Unless -DBUILD_OPTEE_TAS=n, TAs
+# are built.
+#
+# Helper macro define_ta_source_dir() ease integzation of TA source trees.
+#
+# Using macro define_ta_source_dir(TA_NAME TA_SOURCE_PATH)
+# - 'make all' will build TA from source
+# - 'make install' will install TA binary file (*.ta) in /lib/optee_armtz
+# - 'make clean' will clean TA build directory
+if (NOT BUILD_OPTEE_TAS)
+	set (BUILD_OPTEE_TAS y)
+endif ()
+
+macro (define_ta_source_dir TA_NAME TA_SOURCE_PATH)
+
+	if (BUILD_OPTEE_TAS)
+	project (optee_example_${TA_NAME}_ta)
+
+	set_directory_properties(
+		PROPERTIES ADDITIONAL_MAKE_CLEAN_FILES "${CMAKE_CURRENT_SOURCE_DIR}/${TA_SOURCE_PATH}/out"
+		)
+
+	add_custom_target (
+		build-optee_example_${TA_NAME}_ta ALL
```

Comment:
Minor: suffix should be `-ta` instead of `_ta` since you have prefix `build-` and not `build_`